### PR TITLE
Feature/s3 c 3229 bucketnotif objmd update

### PR DESCRIPTION
**Pull request by dora-korpar**
Adds `originOp` key to ObjectMD for `Put`, `Copy`, `CompleteMultipartUpload`, and `DeleteMarkerCreated` operations for bucket notifications

### CODE REVIEW

**Review by miniscruff — APPROVED** [review]

File: lib/api/apiUtils/object/createAndStoreObject.js [hunk 1]
```diff
@@ -148,6 +148,7 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
             removeAWSChunked(request.headers['content-encoding']);
         metadataStoreParams.expires = request.headers.expires;
         metadataStoreParams.tagging = request.headers['x-amz-tagging'];
+        metadataStoreParams.originOp = 's3:ObjectCreated:Put';
```
Comment (naren-rajendran) [inline lib/api/apiUtils/object/createAndStoreObject.js:151]: I am ok with the string literal here, but are you going to maintain a dictionary/list of ops?

File: lib/api/apiUtils/object/createAndStoreObject.js [hunk 2]
```diff
@@ -162,6 +163,7 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         // eslint-disable-next-line no-param-reassign
         request.headers[constants.objectLocationConstraintHeader] =
             objMD[constants.objectLocationConstraintHeader];
+        metadataStoreParams.originOp = 's3:ObjectRemoved:DeleteMarkerCreated';
```
Comment (naren-rajendran) [inline lib/api/apiUtils/object/createAndStoreObject.js:166]: Same here and in the subsequent places.

**Review by naren-rajendran — APPROVED** [review]

File: lib/api/apiUtils/object/createAndStoreObject.js (hunk 1 above)
Comment (dora-korpar) [reply lib/api/apiUtils/object/createAndStoreObject.js:151]: There's a list being added in the Arsenal pr for putBucketNotification

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**